opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java

```diff
@@ -117,70 +117,100 @@ public OnmsAlarm persist(Event event) {
         return alarm;
     }
 
-    private OnmsAlarm addOrReduceEventAsAlarm(Event event) {
-        final OnmsEvent e = m_eventDao.get(event.getDbid());
-        if (e == null) {
+    private OnmsAlarm addOrReduceEventAsAlarm(Event event) throws IllegalStateException {
+        final OnmsEvent persistedEvent = m_eventDao.get(event.getDbid());
+        if (persistedEvent == null) {
             throw new IllegalStateException("Event with id " + event.getDbid() + " was deleted before we could retrieve it and create an alarm.");
         }
 
         final String reductionKey = event.getAlarmData().getReductionKey();
         LOG.debug("addOrReduceEventAsAlarm: looking for existing reduction key: {}", reductionKey);
         OnmsAlarm alarm = m_alarmDao.findByReductionKey(reductionKey);
+        
 
-        if (alarm == null) {
+        boolean newIfCleared = Boolean.getBoolean("opennms.alarmd.newIfClearedAlarmExists");
```

Comment:
This should be declared as a static constant instead of re-reading the system property on every function call - it won't change during the lifetime of the JVM.